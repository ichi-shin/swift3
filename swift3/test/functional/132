#!/bin/bash
# Copyright(c)2014 NTT corp. All Rights Reserved.
# DB1-1-8-1_4

. ./common

S3USER=tester

_s3_put /bucket

upload_id=$(_s3_post /bucket/obj_cache_control?uploads \
    -H "Cache-Control: no-cache" | \
    _xpath '/InitiateMultipartUploadResult/UploadId/text()')

_s3_delete /bucket/obj_cache_control?uploadId=${upload_id}

upload_id=$(_s3_post /bucket/obj_expires?uploads \
    -H "Expires: Wed, 15 Apr 2015 20:00:00 GMT" | \
    _xpath '/InitiateMultipartUploadResult/UploadId/text()')

_s3_delete /bucket/obj_expires?uploadId=${upload_id}


_s3_post /bucket/obj_x-amz-server-side-encryption?uploads \
    -H "x-amz-server-side-encryption: AES256" | _filter_curl xml

_s3_post /bucket/obj_x-amz-website-redirect-location?uploads \
    -H "x-amz-website-redirect-location: http://example.com/" | \
    _filter_curl xml


upload_id=$(_s3_post /bucket/obj_x-amz-grant-write?uploads \
    -H 'x-amz-grant-write: id="handa:handa2"' | \
    _xpath '/InitiateMultipartUploadResult/UploadId/text()')

_s3_delete /bucket/obj_x-amz-grant-write?uploadId=${upload_id}
