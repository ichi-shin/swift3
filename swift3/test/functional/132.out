QA output created by 132
> s3curl --id tester -- -X PUT http://SWIFT_HOST/bucket... 200
> s3curl --id tester -- -X POST -H Cache-Control: no-cache http://SWIFT_HOST/bucket/obj_cache_control?uploads... 200
> s3curl --id tester -- -X DELETE http://SWIFT_HOST/bucket/obj_cache_control?uploadId=UPLOAD_ID... 204
> s3curl --id tester -- -X POST -H Expires: Wed, 15 Apr 2015 20:00:00 GMT http://SWIFT_HOST/bucket/obj_expires?uploads... 200
> s3curl --id tester -- -X DELETE http://SWIFT_HOST/bucket/obj_expires?uploadId=UPLOAD_ID... 204
> s3curl --id tester -- -X POST -H x-amz-server-side-encryption: AES256 http://SWIFT_HOST/bucket/obj_x-amz-server-side-encryption?uploads... 501
<?xml version='1.0' encoding='UTF-8'?>
<Error>
  <Code>NotImplemented</Code>
  <Message>Server-side encryption is not supported.</Message>
  <RequestId>TXID</RequestId>
</Error>
> s3curl --id tester -- -X POST -H x-amz-website-redirect-location: http://example.com/ http://SWIFT_HOST/bucket/obj_x-amz-website-redirect-location?uploads... 501
<?xml version='1.0' encoding='UTF-8'?>
<Error>
  <Code>NotImplemented</Code>
  <Message>Website redirection is not supported.</Message>
  <RequestId>TXID</RequestId>
</Error>
> s3curl --id tester -- -X POST -H x-amz-grant-write: id="handa:handa2" http://SWIFT_HOST/bucket/obj_x-amz-grant-write?uploads... 200
> s3curl --id tester -- -X DELETE http://SWIFT_HOST/bucket/obj_x-amz-grant-write?uploadId=UPLOAD_ID... 204
